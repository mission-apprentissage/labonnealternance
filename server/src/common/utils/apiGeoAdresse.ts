import axios from "axios"

// Cf Documentation : https://geo.api.gouv.fr/adresse
const apiEndpoint = "https://api-adresse.data.gouv.fr"

class ApiGeoAdresse {
  constructor() {}

  async search(q, postcode = null) {
    try {
      const query = `${apiEndpoint}/search/?q=${q ? q : "a"}${postcode ? `&postcode=${postcode}` : ""}`
      let response = await this.searchQuery(query)

      return response
    } catch (error) {
      console.log(`geo search error : ${q} ${postcode} ${error}`)
      return null
    }
  }

  async searchQuery(query) {
    let response
    let trys = 0

    while (trys < 3) {
      response = await axios.get(query)

      if (response?.data?.status === 429) {
        console.log("429 ", new Date(), query)
        trys++
        await new Promise((resolve) => setTimeout(resolve, 1000))
      } else {
        break
      }
    }

    return response?.data
  }

  async searchPostcodeOnly(q, postcode = null) {
    try {
      const response = await axios.get(`${apiEndpoint}/search/?q=${q}${postcode ? `&postcode=${postcode}` : ""}`)
      return response.data
    } catch (error) {
      console.log(`geo searchPostcodeOnly error : #${q}# ${postcode} ${error}`)
      return null
    }
  }
}

const apiGeoAdresse = new ApiGeoAdresse()
export default apiGeoAdresse
