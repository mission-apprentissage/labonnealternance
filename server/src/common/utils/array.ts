/**
 * @description Removes duplicated objects.
 * @param {T[]} arr
 * @param {string[]} properties
 * @returns {T[]}
 */
const getUniqueArray = (arr, properties) => {
  const modifiedArray = []

  if (properties.length === 0 && arr.length > 0) {
    properties.push(...Object.keys(arr[0]))
  }

  arr.map((item) => {
    if (modifiedArray.length === 0) {
      return modifiedArray.push(item)
    }

    if (!modifiedArray.some((item2) => properties.every((property) => item2[property] === item[property]))) {
      return modifiedArray.push(item)
    }
  })

  return modifiedArray
}

export { getUniqueArray }
