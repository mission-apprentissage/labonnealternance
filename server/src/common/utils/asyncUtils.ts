export const asyncForEach = async <T>(array: T[], callback: (item: T, index: number) => Promise<void>) => {
  for (let index = 0; index < array.length; index++) {
    await callback(array[index], index)
  }
}

export const asyncForEachGrouped = async <T>(array: T[], groupSize: number, callback: (item: T, index: number) => Promise<void>) => {
  for (let index = 0; index < array.length; index += groupSize) {
    const group = array.slice(index, index + groupSize)
    await Promise.all(group.map((item, itemIndex) => callback(item, index + itemIndex)))
  }
}

export const delay = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms))

export function timeout(promise, millis) {
  let timeout: NodeJS.Timeout | null = null
  const timeoutPromise = new Promise((resolve, reject) => (timeout = setTimeout(() => reject(`Timed out after ${millis} ms.`), millis)))
  return Promise.race([promise, timeoutPromise]).finally(() => timeout !== null && clearTimeout(timeout))
}

export async function sleep(durationMs: number, signal?: AbortSignal): Promise<void> {
  await new Promise<void>((resolve) => {
    let timeout: NodeJS.Timeout | null = null

    const listener = () => {
      if (timeout) clearTimeout(timeout)
      resolve()
    }

    timeout = setTimeout(() => {
      signal?.removeEventListener("abort", listener)
      resolve()
    }, durationMs)

    signal?.addEventListener("abort", listener)
  })
}
