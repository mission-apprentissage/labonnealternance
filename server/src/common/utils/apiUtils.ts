import { AxiosInstance } from "axios"
import { AxiosCacheInstance } from "axios-cache-interceptor"
import { RateLimiterMemory, RateLimiterQueue } from "rate-limiter-flexible"

import { timeout } from "./asyncUtils"

interface ApiRateLimiterOptions {
  nbRequests?: number
  durationInSeconds?: number
  maxQueueSize?: number
  timeout?: number
  client: AxiosInstance | AxiosCacheInstance
}
export const apiRateLimiter = (name: string, options: ApiRateLimiterOptions) => {
  const rateLimiter = new RateLimiterMemory({
    keyPrefix: name,
    points: options.nbRequests || 1,
    duration: options.durationInSeconds || 1,
  })

  const queue = new RateLimiterQueue(rateLimiter, {
    maxQueueSize: options.maxQueueSize || 25,
  })

  return async <T>(callback: (i: AxiosInstance | AxiosCacheInstance) => T): Promise<T> => {
    await timeout(queue.removeTokens(1), options.timeout || 10000)
    return callback(options.client)
  }
}

export class ApiError extends Error {
  apiName: string
  message: string
  reason: string | undefined
  status: string | undefined

  constructor(apiName: string, message: string, reason?: string, status?: string) {
    super()
    Error.captureStackTrace(this, this.constructor)
    this.name = this.constructor.name
    this.apiName = apiName
    this.message = `[${apiName}] ${message}`
    this.reason = reason
    this.status = status
  }
}
