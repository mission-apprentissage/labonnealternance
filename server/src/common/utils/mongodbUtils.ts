import { captureException } from "@sentry/node"
import { isEqual } from "lodash-es"
import { Collection, CollectionInfo, MongoClient, MongoServerError } from "mongodb" // to uncomment when migrated to V7
import { IModelDescriptor } from "shared/models/common"
import { CollectionName, IDocument, modelDescriptors } from "shared/models/models"
import { zodToMongoSchema } from "zod-mongodb-schema"

import config from "../../config"
import { logger } from "../logger"

import { sleep } from "./asyncUtils"

let mongodbClient: MongoClient | null = null
let mongodbClientState: string | null = null

export const ensureInitialization = () => {
  if (!mongodbClient) {
    throw new Error("Database connection does not exist. Please call connectToMongodb before.")
  }
  return mongodbClient
}

/**
 * @param  {string} uri
 * @returns client
 */
export const connectToMongodb = async (uri: string) => {
  const client = new MongoClient(uri, {
    heartbeatFrequencyMS: 10_000,
    retryWrites: true,
    retryReads: true,
    minPoolSize: config.env === "local" ? 0 : 5,
    maxPoolSize: 1_000,
  })

  client.on("connectionPoolReady", () => {
    logger.info("MongoDB reconnected")
    mongodbClient = client
  })

  client.on("connectionPoolClosed", () => {
    logger.warn("MongoDB closed")
    mongodbClient = null
  })

  await client.connect()
  // @ts-expect-error
  mongodbClientState = client.topology.s.state
  mongodbClient = client
  logger.info("Connected to MongoDB")

  return client
}

export const getMongodbClient = () => mongodbClient
export const getMongodbClientState = () => mongodbClientState

export const closeMongodbConnection = async () => {
  logger.warn("Closing MongoDB")
  // Let 100ms for possible callback cleanup to register tasks in mongodb queue
  await sleep(200)
  return mongodbClient?.close()
}

export const getDatabase = () => {
  return ensureInitialization().db()
}

export const getDbCollection = <K extends CollectionName>(name: K): Collection<IDocument<K>> => {
  return ensureInitialization().db().collection(name)
}

export const getCollectionList = () => {
  return ensureInitialization().db().listCollections().toArray()
}

export const getDbCollectionIndexes = async (name: CollectionName) => {
  return await ensureInitialization().db().collection(name).indexes()
}

/**
 * Création d'une collection si elle n'existe pas
 * @param {string} collectionName
 */
const createCollectionIfDoesNotExist = async (collectionName: string) => {
  const db = getDatabase()
  const collectionsInDb = await db.listCollections().toArray()
  const collectionExistsInDb = collectionsInDb.map(({ name }) => name).includes(collectionName)

  if (!collectionExistsInDb) {
    try {
      await db.createCollection(collectionName)
    } catch (err) {
      if ((err as MongoServerError).codeName !== "NamespaceExists") {
        throw err
      }
    }
  }
}

/**
 * Vérification de l'existence d'une collection à partir de la liste des collections
 * @param {*} collectionsInDb
 * @param {*} collectionName
 * @returns
 */
export const collectionExistInDb = (collectionsInDb: CollectionInfo[], collectionName: string) => collectionsInDb.map(({ name }: { name: string }) => name).includes(collectionName)

/**
 * Config de la validation
 * @param {*} modelDescriptors
 */
export const configureDbSchemaValidation = async (modelDescriptors: IModelDescriptor[]) => {
  const db = getDatabase()
  ensureInitialization()
  await Promise.all(
    modelDescriptors.map(async ({ collectionName, zod }) => {
      await createCollectionIfDoesNotExist(collectionName)

      const convertedSchema = zodToMongoSchema(zod)

      try {
        await db.command({
          collMod: collectionName,
          validationLevel: "strict",
          validationAction: "error",
          validator: {
            $jsonSchema: {
              title: `${collectionName} validation schema`,
              ...convertedSchema,
            },
          },
        })
      } catch (error) {
        captureException(error)
        logger.error(error)
      }
    })
  )
}

/**
 * Clear de toutes les collections
 * @returns
 */
export const clearAllCollections = async () => {
  const collections = await getDatabase().collections()
  return Promise.all(collections.map((c) => c.deleteMany({})))
}

/**
 * Clear d'une collection
 * @param {string} name
 * @returns
 */
export async function clearCollection(name: string) {
  ensureInitialization()
  await getDatabase().collection(name).deleteMany({})
}

export const createIndexes = async () => {
  for (const descriptor of modelDescriptors) {
    if (!descriptor.indexes) {
      return
    }
    const indexes = await getDbCollection(descriptor.collectionName)
      .listIndexes()
      .toArray()
      .catch((err) => {
        // NamespaceNotFound
        if (err.code === 26) {
          return []
        }
        throw err
      })
    const indexesToRemove = new Set(indexes.filter((i) => i.name !== "_id_"))

    logger.info(`Create indexes for collection ${descriptor.collectionName}`)
    await Promise.all(
      descriptor.indexes.map(async ([index, options]): Promise<void> => {
        try {
          const existingIndex =
            // Use Object.entries because order matters
            indexes.find((i) => isEqual(Object.entries(i.key), Object.entries(index)) || options.name === i.name) ?? null

          if (existingIndex) {
            indexesToRemove.delete(existingIndex)
          }

          await getDbCollection(descriptor.collectionName)
            .createIndex(index, options)
            .catch(async (err) => {
              // IndexOptionsConflict & IndexKeySpecsConflict
              if (err.code === 85 || err.code === 86) {
                await getDbCollection(descriptor.collectionName).dropIndex(existingIndex.name)
                await getDbCollection(descriptor.collectionName).createIndex(index, options)
              } else {
                throw err
              }
            })
        } catch (err) {
          captureException(err)
          logger.error(`Error creating indexes for ${descriptor.collectionName}: ${err}`)
        }
      })
    )

    if (indexesToRemove.size > 0) {
      logger.warn(`Dropping extra indexes for collection ${descriptor.collectionName}`, indexesToRemove)
      await Promise.all(Array.from(indexesToRemove).map((index) => getDbCollection(descriptor.collectionName).dropIndex(index.name)))
    }
  }
}

export const dropIndexes = async () => {
  const collections = (await getCollectionList()).map((collection) => collection.name)
  for (const descriptor of modelDescriptors) {
    logger.info(`Drop indexes for collection ${descriptor.collectionName}`)
    if (collections.includes(descriptor.collectionName)) {
      await getDbCollection(descriptor.collectionName).dropIndexes()
    }
  }
}
