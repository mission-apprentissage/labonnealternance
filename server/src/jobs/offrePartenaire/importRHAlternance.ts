import { internal } from "@hapi/boom"
import axios from "axios"
import { ObjectId } from "mongodb"
import { IRawRHAlternance } from "shared/models/rawRHAlternance.model"
import { z } from "zod"

import { logger } from "@/common/logger"
import { getDbCollection } from "@/common/utils/mongodbUtils"
import config from "@/config"

const ZRHAlternanceResponse = z
  .object({
    jobs: z.array(z.unknown()),
    pageCount: z.number(),
  })
  .passthrough()

const destinationCollection = "raw_rhalternance"

export const importRHAlternance = async () => {
  logger.info("deleting old data")
  await getDbCollection(destinationCollection).deleteMany({})
  logger.info("import starting...")

  let shouldContinue = true
  let currentPage = 1
  let pageCount: number | null = null
  let jobCount = 0
  const importDate = new Date()
  while (shouldContinue) {
    logger.info(`import of RHAlternance: page ${currentPage} / ${pageCount ?? "?"}`)
    const response = await axios.request({
      method: "POST",
      url: "https://rhalternance.com/api/jobs",
      headers: {
        Authorization: config.rhalternance.apiKey,
      },
      data: `Page=${currentPage}`,
    })
    if (response.status >= 400) {
      throw internal(`Error when calling RHAlternance: status=${response.status}`, { data: response.data })
    }
    const parsed = ZRHAlternanceResponse.safeParse(response.data)
    if (!parsed.success) {
      throw internal(`Error when parsing RHAlternance response`, { error: parsed.error })
    }
    const savedJobs: IRawRHAlternance[] = parsed.data.jobs.map((job) => ({ job, createdAt: importDate, _id: new ObjectId() }))
    if (savedJobs.length) {
      await getDbCollection(destinationCollection).insertMany(savedJobs)
      currentPage++
      pageCount = parsed.data.pageCount
    } else {
      shouldContinue = false
    }
    jobCount += savedJobs.length
  }
  logger.info(`import done: ${jobCount} jobs imported`)
}
