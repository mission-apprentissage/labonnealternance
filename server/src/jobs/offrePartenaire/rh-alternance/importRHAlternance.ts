import { internal } from "@hapi/boom"
import axios from "axios"
import { ObjectId } from "mongodb"
import { TRAINING_CONTRACT_TYPE } from "shared/constants"
import { JOBPARTNERS_LABEL } from "shared/models/jobsPartners.model"
import { IComputedJobsPartners } from "shared/models/jobsPartnersComputed.model"
import rawRHAlternanceModel, { IRawRHAlternance } from "shared/models/rawRHAlternance.model"
import { joinNonNullStrings } from "shared/utils"
import { z } from "zod"

import { logger } from "@/common/logger"
import { getDbCollection } from "@/common/utils/mongodbUtils"
import { notifyToSlack } from "@/common/utils/slackUtils"
import config from "@/config"
import dayjs from "@/services/dayjs.service"

import { blankComputedJobPartner } from "../fillComputedJobsPartners"
import { rawToComputedJobsPartners } from "../rawToComputedJobsPartners"

const ZRawRHAlternanceJob = rawRHAlternanceModel.zod.shape.job

const ZRHAlternanceResponse = z
  .object({
    jobs: z.array(ZRawRHAlternanceJob),
    pageCount: z.number(),
  })
  .passthrough()

const rawCollectionName = rawRHAlternanceModel.collectionName

export const importRHAlternanceRaw = async () => {
  logger.info("deleting old data")
  await getDbCollection(rawCollectionName).deleteMany({})
  logger.info("import starting...")

  let shouldContinue = true
  let currentPage = 1
  let pageCount: number | null = null
  let jobCount = 0
  const importDate = new Date()
  while (shouldContinue) {
    logger.info(`import of RHAlternance: page ${currentPage} / ${pageCount ?? "?"}`)
    const response = await axios.request({
      method: "POST",
      url: "https://rhalternance.com/api/jobs",
      headers: {
        Authorization: config.rhalternance.apiKey,
      },
      data: `Page=${currentPage}`,
    })
    if (response.status >= 400) {
      throw internal(`Error when calling RHAlternance: status=${response.status}`, { data: response.data })
    }
    const parsed = ZRHAlternanceResponse.safeParse(response.data)
    if (!parsed.success) {
      throw internal(`Error when parsing RHAlternance response`, { error: parsed.error })
    }
    const savedJobs: IRawRHAlternance[] = parsed.data.jobs.map((job) => ({ job, createdAt: importDate, _id: new ObjectId() }))
    if (savedJobs.length) {
      await getDbCollection(rawCollectionName).insertMany(savedJobs)
      currentPage++
      pageCount = parsed.data.pageCount
    } else {
      shouldContinue = false
    }
    jobCount += savedJobs.length
  }
  const message = `import RH Alternance terminé : ${jobCount} offres importées`
  logger.info(message)
  await notifyToSlack({
    subject: `import des offres RH Alternance dans raw`,
    message,
  })
}

export const importRHAlternanceToComputed = async () => {
  const now = new Date()
  await rawToComputedJobsPartners({
    collectionSource: rawCollectionName,
    partnerLabel: JOBPARTNERS_LABEL.RH_ALTERNANCE,
    zodInput: ZRawRHAlternanceJob,
    mapper: rawRhAlternanceToComputedMapper(now),
    documentJobRoot: "job",
  })
}

export const rawRhAlternanceToComputedMapper =
  (now: Date) =>
  ({
    jobCode,
    companyName,
    companySiret,
    companyUrl,
    jobTitle,
    jobDescription,
    jobSubmitDateTime,
    jobType,
    jobUrl,
    jobCity,
    jobPostalCode,
  }: IRawRHAlternance["job"]): IComputedJobsPartners => {
    const offer_creation = jobSubmitDateTime ? dayjs.tz(jobSubmitDateTime).toDate() : now
    const isValid: boolean = jobType === "Alternance"
    const computedJob: IComputedJobsPartners = {
      ...blankComputedJobPartner,
      _id: new ObjectId(),
      partner_job_id: jobCode,
      partner_label: JOBPARTNERS_LABEL.RH_ALTERNANCE,
      contract_type: jobType === "Alternance" ? [TRAINING_CONTRACT_TYPE.APPRENTISSAGE, TRAINING_CONTRACT_TYPE.PROFESSIONNALISATION] : [],
      offer_title: jobTitle,
      offer_description:
        (jobDescription ?? [])
          .map(({ descriptionHeadline, descriptionText }) => [descriptionHeadline, descriptionText].join(" : "))
          .filter((line) => line.length)
          .join("\n")
          .replace(/<br\s*\/?>/g, "\r\n") || null,
      offer_creation,
      offer_expiration: dayjs.tz(offer_creation).add(60, "days").toDate(),
      offer_opening_count: 1,
      offer_multicast: true,
      workplace_siret: companySiret,
      workplace_name: companyName,
      workplace_website: companyUrl,
      workplace_address_label: joinNonNullStrings([jobPostalCode, jobCity]),
      workplace_address_city: jobCity,
      workplace_address_zipcode: jobPostalCode,
      apply_url: jobUrl,
      business_error: isValid ? null : `expected jobType === "Alternance" but got ${jobType}`,
      updated_at: now,
    }
    return computedJob
  }
