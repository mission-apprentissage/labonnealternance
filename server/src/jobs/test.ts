import { ObjectId } from "bson"
import { IApplicant, IApplication, ZApplicant } from "shared"

import { getDbCollection } from "@/common/utils/mongodbUtils"

import { asyncForEach } from "../common/utils/asyncUtils"

type ApplicationAggregate = Pick<IApplication, "applicant_email" | "applicant_first_name" | "applicant_last_name" | "applicant_phone" | "_id">

export const migrationTest = async () => {
  // console.log(extensions.telephone.safeParse("0690760552"))
  // return
  const applications = (await getDbCollection("applications")
    .find({}, { projection: { applicant_first_name: 1, applicant_last_name: 1, applicant_email: 1, applicant_phone: 1, _id: 1 } })
    .toArray()) as ApplicationAggregate[]

  const stat = { error: 0, success: 0, total: applications.length }

  await asyncForEach(applications, async (application) => {
    const { applicant_first_name, applicant_email, applicant_last_name, applicant_phone, _id } = application
    const now = new Date()
    const applicant: IApplicant = {
      _id: new ObjectId(),
      firstname: applicant_first_name,
      lastname: applicant_last_name,
      phone: applicant_phone,
      email: applicant_email,
      last_connection: null,
      createdAt: now,
      updatedAt: now,
    }
    const validation = ZApplicant.safeParse(applicant)
    if (!validation.success) {
      stat.error++
      console.log({ applicant, err: validation.error.errors })
    } else {
      stat.success++
    }
  })
  console.log(stat)
}
