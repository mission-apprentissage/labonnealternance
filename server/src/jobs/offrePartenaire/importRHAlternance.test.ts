import { useMongo } from "@tests/utils/mongo.test.utils"
import { beforeEach, describe, expect, it, vi } from "vitest"

import { getDbCollection } from "@/common/utils/mongodbUtils"

import { generateRawRHAlternanceJobFixture } from "../../../../shared/fixtures/rawRHAlternanceJob.fixture"

import { rawRhAlternanceToComputedMapper } from "./importRHAlternance"

const now = new Date("2024-07-21T04:49:06.000+02:00")

describe("import RH Alternance", () => {
  useMongo()

  beforeEach(() => {
    vi.useFakeTimers()
    vi.setSystemTime(now)

    return async () => {
      vi.useRealTimers()
      await getDbCollection("computed_jobs_partners").deleteMany({})
      await getDbCollection("raw_rhalternance").deleteMany({})
    }
  })

  describe("rawRhAlternanceToComputedMapper", () => {
    it("should test that the mapper works", async () => {
      const mapped = rawRhAlternanceToComputedMapper(now)(generateRawRHAlternanceJobFixture())
      expect.soft(mapped.business_error).toBeFalsy()
      expect.soft(mapped).toMatchSnapshot()
    })
    it("should detect a business error", async () => {
      const mapped = rawRhAlternanceToComputedMapper(now)(
        generateRawRHAlternanceJobFixture({
          jobType: "invalid",
        })
      )
      expect.soft(mapped.business_error).toBeTruthy()
    })
  })
})
