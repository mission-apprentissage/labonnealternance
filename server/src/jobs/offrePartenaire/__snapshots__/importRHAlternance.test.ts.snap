// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`import RH Alternance > rawRhAlternanceToComputedMapper > should test that the mapper works 1`] = `
{
  "apply_phone": null,
  "apply_url": "https://rhalternance.com/jobs/environnement/la-grande-motte/alternant-hse-h-f-913079004",
  "business_error": null,
  "contract_duration": null,
  "contract_remote": null,
  "contract_start": null,
  "contract_type": [
    "Apprentissage",
    "Professionnalisation",
  ],
  "created_at": null,
  "errors": [],
  "jobs_in_success": [],
  "offer_access_conditions": [],
  "offer_creation": 2024-11-14T11:11:08.000Z,
  "offer_description": "Détail du poste : QUE FEREZ-VOUS ?<br><br>Rattaché(e) à Arnaud, Responsable HSE Outremer, vous contribuerez au déploiement d'une démarche HSE amenant les équipes vers une évolution de la maturité HSE de l'entreprise.<br><br>Vos missions principales seront :<br><br>* Participer au déploiement de la feuille de route HSE 2024-25,<br><br>* Participer à l'animation de la culture HSE dans l'établissement (1/4 sécurité/ accueil sécurité / sensibilisation quotidienne, communication «Safety First »...),<br><br>* La mise à jour de l'évaluation des risques professionnels et risques d'exposition,<br><br>* La mise en place des indicateurs et tableaux de bord,<br><br>* Veille documentaires (rédaction de documents / consignes/ mise à jour des FDS / process),<br><br>* Participer à l'analyse environnementale du site et proposition d'un plan d'action associé,<br><br>* La recherche de réduction de nos impacts, comme la réduction et/ou la valorisation de nos déchets (emballages plastiques, cartons...).<br><br>NOUS SOMMES FIERS DE POUVOIR PROPOSER DEPUIS DES ANNÉES DES PARCOURS D'ÉVOLUTION À NOS SALARIÉS.
Le profil recherché : CE POSTE EST FAIT POUR VOUS SI VOUS :<br><br>* Vous êtes en Licence ou Master, cursus QHSE avec idéalement avec une expérience dans l'industrie (stage ou alternance),<br><br>* Vous avez connaissance des réglementations HSE, des systèmes de management QHSE,<br><br>* Vous êtes force de proposition et d'amélioration, êtes méthodique et rigoureux(se),<br><br>* Vous êtes bon communiquant et êtes à l'aise en milieu industriel,<br><br>* Vous êtes à l'aise avec les outils informatiques.<br><br>COMMENT ÇA SE PASSE ?<br><br>Envoyez-nous votre CV. Si votre profil correspond à notre besoin, vous serez contacté(e) par téléphone et accueilli(e) à la Grande Motte pour visiter le chantier et rencontrer nos équipes.<br><br>Chez Grand Large Yachting, nous avons à coeur d'exclure toute forme de discrimination dans le cadre de nos recrutements.
Infos complémentaires : Nous vous proposons :<br><br>Un poste en alternance ouvert dès Septembre 2024 à La Grande Motte (34) à côté du port.<br>- Rémunération fixe,<br>- Primes de production,<br>- Participation aux bénéfices de l'entreprise<br>- Mutuelle intéressante pour vous et toute la famille (prise en charge employeur à 50%),<br>- Prime mobilité durable (aide pour le trajet domicile/travail si utilisation de vélo, trottinette, covoiturage),<br><br>- Nos petits plus :<br>- CSE (chèques cadeaux/vacances et divers autres avantages),<br>- Team Day,<br>- Activités nautiques en partenariat avec l'école de voile de la Grande Motte au printemps.",
  "offer_desired_skills": [],
  "offer_expiration": 2025-01-13T11:11:08.000Z,
  "offer_multicast": true,
  "offer_opening_count": 1,
  "offer_origin": null,
  "offer_rome_codes": null,
  "offer_status": "Active",
  "offer_target_diploma": null,
  "offer_title": "Alternant - HSE H/F",
  "offer_to_be_acquired_skills": [],
  "partner_job_id": "913079004",
  "partner_label": "RH Alternance",
  "updated_at": 2024-07-21T02:49:06.000Z,
  "validated": false,
  "workplace_address_city": "La Grande-Motte",
  "workplace_address_label": "34280 La Grande-Motte",
  "workplace_address_street_label": null,
  "workplace_address_zipcode": "34280",
  "workplace_brand": null,
  "workplace_description": null,
  "workplace_geopoint": null,
  "workplace_idcc": null,
  "workplace_legal_name": null,
  "workplace_naf_code": null,
  "workplace_naf_label": null,
  "workplace_name": "GRAND LARGE YACHTING",
  "workplace_opco": null,
  "workplace_siret": "44537545400037",
  "workplace_size": null,
  "workplace_website": "https://rhalternance.com/companies/grand-large-yachting-mediterranee",
}
`;
